modules/@apostrophecms/image/ui/apos/components/AposMediaManager.vue

```diff
@@ -367,26 +351,33 @@ export default {
               draft: true
             }
           ));
-          if (!this.mounted) {
-            return;
-          }
-          this.tagList = apiResponse.choices._tags;
+          result.tagList = apiResponse.choices._tags;
         } else {
-          this.tagList = apiResponse.choices ? apiResponse.choices._tags : [];
+          result.tagList = apiResponse.choices ? apiResponse.choices._tags : [];
         }
       }
 
-      this.currentPage = apiResponse.currentPage;
-      this.totalPages = apiResponse.pages;
+      result.currentPage = apiResponse.currentPage;
+      result.totalPages = apiResponse.pages;
+      result.items = [];
       for (const image of apiResponse.results) {
-        this.items.push(image);
+        result.items.push(image);
+      }
+      return result;
+    },
+    async appendMedia({ tagList, currentPage, totalPages, items }) {
+      this.tagList = tagList;
+      this.currentPage = currentPage;
+      this.totalPages = totalPages;
+      for (const item of items) {
+        this.items.push(item);
       }
     },
     async refetchMedia(opts) {
       this.isLoading = true;
       this.currentPage = 1;
       this.items = [];
-      await this.getMedia(opts);
+      await this.debouncedGetMedia.skipDelay(opts);
       this.isLoading = false;
```

Comment:
No need for try/catch because in the presence of `ifNotCanceled` the wrapper always resolves to null regardless of which way "fn" settles (if it even gets called). However it does wait for "fn" to settle if it is called so we can do this kind of cleanup easily.

(I don't know if setting `isLoading` only some of the time is a bug or a feature, I assume it's a feature. Otherwise it would have been moved inside getMedia itself by now)